kliff/models/kim.py

```diff
@@ -576,21 +540,58 @@ def set_one_opt_param(self, name: str, settings: List[List[Any]]):
 
         Args:
             name: name of a fitting parameter
-            settings: initial value, flag to fix a parameter, lower and upper bounds of a
+            settings: List initial value, flag to fix a parameter, lower and upper bounds of a
                 parameter.
 
         Example:
-            name = 'param_A'
+            name = 'A'
             settings = [['default', 0, 20], [2.0, 'fix'], [2.2, 'inf', 3.3]]
             instance.set_one(name, settings)
         """
-        self.opt_params.set_one(name, settings)
+        # Keeping this API intact for now, no improvement comes to mind
+        # self.opt_params.set_one(name, settings)
+        param = self.model_params[name]
+        # check the val kind
+        supplied_value = settings[0][0]
+        if supplied_value == "default":
+            supplied_value = param.get_numpy_array_model_space()
+        elif isinstance(supplied_value, (int, float)):
+            supplied_value = np.array([supplied_value])
+        elif isinstance(supplied_value, Parameter):
+            supplied_value = supplied_value.get_numpy_array_model_space()
+        else:
+            raise ValueError("Settings array is not properly formatted")
+        # When model is operating with transformed parameters
+        # input is expected in transformed space
+        param.copy_from_param_space(supplied_value)
```

Comment:
`supplied_value` are in `model space`. Then we shouldn't call `param.copy_from_param_space(supplied_value)` here? 

Also, this is inconsistent with what is in `model.py`, where `get_numpy_array_param_space()` is used.

Reply:
I was under impression that in `set_opt_param` interface supplied values will be in `param_space` as before in yonathan's code, that is where it was transformed. If not, I can convert it to `_model_space` functions.

Reply:
Fixed everywhere to `_param_space`, as I believe we had a  discussion with Yonathan where in case of uq work, the `set_opt_params` provides transformed space values. All required `set_opt_param` and `_one_opt_param` functions now call `_param_space` function. Also added brief note in documentation, explaining designed use of each function in Parameter class.